activetable/activetable.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""An XBlock with a tabular problem type that requires students to fill in some cells."""
+from __future__ import absolute_import, division, unicode_literals
+
+import textwrap
+
+from xblock.core import XBlock
+from xblock.fields import Dict, Float, Integer, Scope, String
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from .cells import NumericCell
+from .parsers import ParseError, parse_table, parse_number_list
+
+loader = ResourceLoader(__name__)  # pylint: disable=invalid-name
+
+
+class ActiveTableXBlock(StudioEditableXBlockMixin, XBlock):
+    """An XBlock with a tabular problem type that requires students to fill in some cells."""
+
+    display_name = String(
+        display_name='Display Name',
+        help='The title Studio uses for the component.',
+        scope=Scope.settings,
+        default='ActiveTable problem'
+    )
+    table_definition = String(
+        display_name='Table definition',
+        help='The definition of the table in Python-like syntax.',
+        scope=Scope.content,
+        multiline_editor=True,
+        resettable_editor=False,
+        default=textwrap.dedent("""\
+        [
+            ['Column header 1', 'Column header 2'],
+            ['Enter "answer" here:', String(answer='answer')],
+            [42, Numeric(answer=42, tolerance=0.0)],
+        ]
+        """)
+    )
+    help_text = String(
+        display_name='Help text',
+        help='The text that gets displayed when clicking the "+help" button.  If you remove the '
+        'help text, the help feature is disabled.',
+        scope=Scope.content,
+        multiline_editor=True,
+        resettable_editor=False,
+        default='Fill in the cells highlighted in yellow with the correct answers.  '
+        'When you are done, you can check your answers using the button below the table.',
+    )
+    column_widths = String(
+        display_name='Column widths',
+        help='Set the width of the columns in pixels.  The value should be a Python-like list of '
+        'numerical values.  The total width of the table should not be more than 800. No value '
+        'will result in equal-width columns with a total width of 800 pixels.',
+        scope=Scope.content,
+    )
+    row_heights = String(
```

Comment:
The row heights and column widths fields have a reset button next to them in the interface. However, this button does not seem to work. Clicking it makes the field appear empty, but saving it does not actually save an empty string-- it just reverts to whatever it was before you hit the button. Putting in a new value does get saved, though.